scripts/install-worker.sh

```diff
@@ -328,6 +341,31 @@ if [[ ! $KUBERNETES_VERSION =~ "1.19"* && ! $KUBERNETES_VERSION =~ "1.20"* && !
     sudo mv $TEMPLATE_DIR/ecr-credential-provider-config /etc/eks/ecr-credential-provider/ecr-credential-provider-config
 fi
 
+################################################################################
+### Cache Images ##############################################################
+################################################################################
+AWS_DOMAIN=$(curl -s http://169.254.169.254/latest/meta-data/services/domain)
```

Comment:
If SCP enforced [IMDSv2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) policy applied to the Org (or IAM policy enforced), it will fail to retrieve metadata from IMDSv1. Need to implement a logic to detect if it running with v1 or v2.

Reply:
👍 we should absolutely be using IMDSv2 everywhere we can. I believe there are some other locations that still use v1 in the building scripts. I'll take an action to update those with this PR.

Reply:
https://github.com/awslabs/amazon-eks-ami/pull/1013